### PR TITLE
Install check for content outside mount point

### PR DESCRIPTION
**Pull request by seancpeters**
Also adds basic trace logging for install & authoring issues (more to come in future PRs)

TODO: Determine exactly what we want to do for localizing the trace messages. All currently localized messages are in the CLI assembly, but these are not.

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs [hunk 1]
```diff
@@ -168,6 +168,50 @@ public IList<ITemplate> GetTemplatesAndLangpacksFromDir(IMountPoint source, out
...
+                try
+                {
+                    IDirectory sourceRoot = runnableTemplate.TemplateSourceRoot.DirectoryInfo(source.Source);
+
+                    if (!sourceRoot.Exists)
+                    {   // non-existant directory
+                        allSourcesValid = false;
+                        host.LogDiagnosticMessage(string.Empty, "Authoring");
+                        host.LogDiagnosticMessage(string.Format("Template: '{0}'", runnableTemplate.Name), "Authoring");
+                        host.LogDiagnosticMessage(string.Format("\tTemplateSourceRoot = '{0}'", runnableTemplate.TemplateSourceRoot.FullPath), "Authoring");
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs]: It'd probably be good to extract these to LocalizableStrings

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs [hunk 2]
```diff
@@ -168,6 +168,50 @@ public IList<ITemplate> GetTemplatesAndLangpacksFromDir(IMountPoint source, out
...
+                        host.LogDiagnosticMessage(string.Format("\tSource '{0}' in template does not exist.", source.Source), "Authoring");
+                        host.LogDiagnosticMessage(string.Format("\tSource path relative to install location: '{0}'", sourceRoot.FullPath), "Authoring");
+                    }
+                }
+                catch
+                {   // outside the mount point root
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs]: We should leave a note to revisit this if we fix the NullRef that happens looking up DirectoryInfo when it escapes the root

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.1.x/content/ClassLibrary-VisualBasic/Class1.vb [hunk 3]
```diff
@@ -1,3 +1,3 @@
-Public Class Class1
```
Comment (mlorbetske) [inline template_feed/Microsoft.DotNet.Common.ProjectTemplates.1.x/content/ClassLibrary-VisualBasic/Class1.vb]: We should get rid of the line ending/encoding changes

File: test/Microsoft.TemplateEngine.EndToEndTestHarness/test_templates/TemplateWithSourceNameAndCustomSourcePath/Content/.template.config/template.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  "name": "TemplateWithSourceNameAndCustomSourcePath",
+  "generatorVersions": "[1.0.0.0-*)",
+  "groupIdentity": "TestAssets.TemplateWithSourceNameAndCustomSourcePath",
+  "precedence": "100",
+  "identity": "TestAssets.TemplateWithSourceNameAndCustomSourcePath",
+  "shortName": "TestAssets.TemplateWithSourceNameAndCustomSourcePath",
+  "sourceName": "foo",
+  "sources": [
+    {
+      //"source": "./Custom/Path/"
```
Comment (mlorbetske) [inline test/Microsoft.TemplateEngine.EndToEndTestHarness/test_templates/TemplateWithSourceNameAndCustomSourcePath/Content/.template.config/template.json]: needed?

Comment (seancpeters) [reply test/Microsoft.TemplateEngine.EndToEndTestHarness/test_templates/TemplateWithSourceNameAndCustomSourcePath/Content/.template.config/template.json]: no, removing

**Comment by seancpeters** [comment]
@dotnet-bot test Ubuntu x64 Release Build please
@dotnet-bot test RHEL7.2 x64 Release Build please

**Comment by seancpeters** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Pull request merged**